### ISSUE TITLE
Typage de + polymorphe un peu long

### ISSUE THREAD

**Issue by chambart** [bvaugon/GADT-implem#1]
Pour
```ocaml
let f x = x + x
```
pas de problème, mais le type est un peu gros quand même...

Pour
```ocaml
let g x = x + x + x
```
c'est un peu trop...

**Comment by bvaugon** [comment]
Oui, effectivement. Comme (+) est surchargé, déjà, ça fait des schémas assez gros. Mais la chose importante est que cette implémentation n'a pas l'optimisation n°4 dont j'ai parlé à ma soutenance, à savoir la factorisation des sous-ensembles de contraintes "équivalents". Avec cette optimisation, ces exemples passent sans problème et le schéma généré est systématiquement le même.
